zipline-loader/src/jvmTest/kotlin/app/cash/zipline/loader/HttpFetcherTest.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.loader
+
+import app.cash.zipline.loader.fetcher.HttpFetcher
+import kotlin.test.assertEquals
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import org.junit.Test
+
+class HttpFetcherTest {
+  private val httpFetcher = HttpFetcher(FakeZiplineHttpClient())
+  private val json = Json {
+    prettyPrint = true
+  }
+
+  @Test
+  fun happyPath() {
+    val manifestWithRelativeUrls =
+      """
+      |{
+      |    "modules": {
+      |        "./hello.js": {
+      |            "url": "hello.zipline",
+      |            "sha256": "6bd4baa9f46afa62477fec8c9e95528de7539f036d26fc13885177b32fc0d6ab"
+      |        },
+      |        "./jello.js": {
+      |            "url": "jello.zipline",
+      |            "sha256": "7af37185091e22463ff627686aedfec3528376eb745026fae1d6153688885e73"
+      |        }
+      |    }
+      |}
+      """.trimMargin()
+
+    val manifestWithResolvedUrls = httpFetcher.resolveUrls(
+      manifest = json.parseToJsonElement(manifestWithRelativeUrls),
+      baseUrl = "https://example.com/path/",
+    )
+
+    assertEquals(
+      """
+      |{
+      |    "modules": {
+      |        "./hello.js": {
+      |            "url": "https://example.com/path/hello.zipline",
```

Comment:
nice use of snapshot tests here to clarify the changes to the entire manifest at a glance